dotnet/aspnet/mssqldbWithTests/Application/SampleWebApplication.FunctionalTests/SampleFunctionalTests.cs

```diff
@@ -18,23 +19,43 @@ public static void Initialize(TestContext testContext)
             SampleFunctionalTests.testContext = testContext;
         }
 
+        [TestInitialize]
+        public void TestInit()
+        {
+            driver = GetChromeDriver();
+        }
+
+        [TestCleanup]
+        public void TestClean()
+        {
+            driver.Quit();
+        }
+
         [TestMethod]
         public void SampleFunctionalTest1()
         {
-            try
-            {
-                driver = GetChromeDriver();
-                var webAppUrl = testContext.Properties["webAppUrl"].ToString();
-                driver.Navigate().GoToUrl(webAppUrl);
-                Assert.AreEqual("Home Page - My ASP.NET Application", driver.Title, "Expected title to be 'Home Page - My ASP.NET Application'");
-                var filePath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString()) + ".png";
-                var screenshot = driver.GetScreenshot();
-                screenshot.SaveAsFile(filePath);
-                testContext.AddResultFile(filePath);
-            }
-            finally
+            var numRetries = 5;
+            for (int i = 0; i < numRetries; i++)
             {
-                driver.Quit();
+                try
+                {
+                    var webAppUrl = testContext.Properties["webAppUrl"].ToString();
+                    driver.Navigate().GoToUrl(webAppUrl);
```

Comment:
Should there be an exception only in this line

Reply:
Added retry on this line as well as the title check